### PR TITLE
feat: fully remove Log Analysis and Activity Tracker support

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Description

Issue: https://github.ibm.com/GoldenEye/issues/issues/11114

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

- Stop provisioning Log analysis instance.
- provision cloud logs instance in examples
- Remove support for installing log analysis agents in the ocp cluster.
- Add support for logs agent installation in the ocp cluster.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Review by Ak-sky — APPROVED** [review]
LGTM

File: variables.tf [hunk 1]
```diff
@@ -369,95 +369,109 @@ variable "custom_security_group_ids" {
...
+    value = optional(string)
   }))
-  default = [{
-    operator = "Exists"
-  }]
+  default = []
+}
+
+variable "cloud_logs_ingress_endpoint" {
+  description = "The host for IBM Cloud Logs ingestion. Ensure you use the ingress endpoint. See https://cloud.ibm.com/docs/cloud-logs?topic=cloud-logs-endpoints_ingress."
```
Comment (ocofaigh) [inline variables.tf:466]: I guess this should say required if `logs_agent_enabled` is set to true. And add validation for this too

File: variables.tf [hunk 2]
```diff
@@ -369,95 +369,109 @@ variable "custom_security_group_ids" {
...
+  description = "The namespace where the Logs agent is deployed. The default value is `ibm-observe`."
+  default     = "ibm-observe"
+  nullable    = false
 }
 
-variable "log_analysis_ingestion_key" {
+variable "logs_agent_iam_api_key" {
   type        = string
-  description = "Ingestion key for the Log Analysis agent to communicate with the instance."
+  description = "The IBM Cloud API key for the Logs agent to authenticate and communicate with the IBM Cloud Logs. It is required if `logs_agent_iam_mode` is set to `IAMAPIKey`."
```
Comment (ocofaigh) [inline variables.tf:397]: required if `logs_agent_enabled` is true and `logs_agent_iam_mode` is set to `IAMAPIKey`. Perhaps add validation for this

File: variables.tf [hunk 3]
```diff
@@ -369,95 +369,109 @@ variable "custom_security_group_ids" {
...
 }
 
-variable "log_analysis_agent_namespace" {
+variable "logs_agent_iam_mode" {
   type        = string
-  description = "Namespace where to deploy the Log Analysis agent. Default value is 'ibm-observe'"
-  default     = "ibm-observe"
-  nullable    = false
+  default     = "TrustedProfile"
+  description = "IAM authentication mode: `TrustedProfile` or `IAMAPIKey`."
```
Comment (ocofaigh) [inline variables.tf:447]: This should mention that if `TrustedProfile` is selected, the module will create one. Mention the exact scope its created with too.
We probably should also support using existing trusted profile too (but I guess can be added later)

**Review by ocofaigh — CHANGES_REQUESTED** [review]
left a few comments

File: variables.tf (hunk 1 above)
Comment (Aashiq-J) [reply variables.tf:466]: Screenshot 2024-10-21 at 4 06 30 PM
There is already a validation present in the obs agent module.

File: variables.tf (hunk 2 above)
Comment (Aashiq-J) [reply variables.tf:397]: Screenshot 2024-10-21 at 4 08 40 PM

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**